src/Kiota.Builder/Writers/Go/GoConventionService.cs

```diff
@@ -0,0 +1,134 @@
+
+        public string HttpCorePropertyName => "httpCore";
+
+        public string ParseNodeInterfaceName => "ParseNode";
+
+        public object AbstractionsHash => "ida96af0f171bb75f894a4013a6b3146a4397c58f11adb81a2b7cbea9314783a9";
+        internal string DocCommentStart = "/*";
+        internal string DocCommentEnd = " */";
+
+        public string GetAccessModifier(AccessModifier access)
+        {
+            throw new InvalidOperationException("go uses a naming convention for access modifiers");
+        }
+        public string GetParameterSignature(CodeParameter parameter) {
+            throw new InvalidOperationException("go needs import symbols, use the local override instead");
+        }
+        public string GetParameterSignature(CodeParameter parameter, CodeElement targetElement)
+        {
+            return $"{parameter.Name} {GetTypeString(parameter.Type, targetElement)}";
+        }
+        public string GetTypeString(CodeTypeBase code) => throw new InvalidOperationException("go needs import symbols, use the local override instead");
+        public string GetTypeString(CodeTypeBase code, CodeElement targetElement, bool addPointerSymbol = true, bool addCollectionSymbol = true)
+        {
+            if(code is CodeUnionType) 
+                throw new InvalidOperationException($"Go does not support union types, the union type {code.Name} should have been filtered out by the refiner");
+            else if (code is CodeType currentType) {
+                var importSymbol = GetImportSymbol(code, targetElement);
+                if(!string.IsNullOrEmpty(importSymbol))
+                    importSymbol += ".";
+                var typeName = TranslateType(currentType.Name);
+                var nullableSymbol = addPointerSymbol && 
+                                    currentType.IsNullable &&
+                                    currentType.CollectionKind == CodeTypeBase.CodeTypeCollectionKind.None &&
+                                    !IsScalarType(currentType.Name) ? "*"
+                                    : string.Empty;
+                var collectionPrefix = currentType.CollectionKind switch {
+                    (CodeType.CodeTypeCollectionKind.Array or CodeType.CodeTypeCollectionKind.Complex) when addCollectionSymbol => "[]",
+                    _ => string.Empty,
+                };
+                if (currentType.ActionOf)
+                    return $"func (value {nullableSymbol}{collectionPrefix}{importSymbol}{typeName}) (err error)";
+                else
+                    return $"{nullableSymbol}{collectionPrefix}{importSymbol}{typeName}";
+            }
+            else throw new InvalidOperationException($"type of type {code.GetType()} is unknown");
+        }
+
+        public string TranslateType(string typeName)
+        {
+            if(typeName.StartsWith("map[")) return typeName; //casing hack
+
+            return (typeName) switch {
```

Comment:
Side note: We may eventually need to think about the `json` type as this miht depend on the serialization representation of the json.

Reply:
Can you expand on that? Any payload should be described as an object/enum/array etc in the openAPI description, I'm not sure how we'd endup with a JSON type if the openAPI description is correct and abstracts away the OData concepts.

Reply:
The metadata description has a ComplexType called Json as below(both V1 and beta).
![image](https://user-images.githubusercontent.com/6464005/129765539-d96ff78a-539f-4743-96ad-79415db92e50.png)

My understanding is that a property with this type would therefore need to be a json representation in the specific language. So, it could potentially be any type of json e.g. object or array and therefore the respective language will probably need to represent it in its appropriate Json Object representation (like maybe JsonDocument or JObject in C#).

A good example with an API like this is this workbooks api [here](https://docs.microsoft.com/en-us/graph/api/range-update?view=graph-rest-1.0&tabs=http#request-body). The `formulas` and `values` properties are declared as `Json`. But the example shows them being sent as Json Arrays while if you dig deeper, you find other api examples with the same type but sent as Json Objects.